sstables/random_access_reader.cc

```diff
@@ -24,6 +24,22 @@ future <temporary_buffer<char>> random_access_reader::read_exactly(size_t n) noe
   }
 }
 
+future <utils::chunked_vector<temporary_buffer<char>>> random_access_reader::read_exactly_fragmented(size_t n) noexcept {
+  utils::chunked_vector<temporary_buffer<char>> result;
+  while (n > 0) {
+    constexpr size_t max_chunk_size = 128*1024;
+    size_t to_read = std::min(n, max_chunk_size);
+    try {
+      result.push_back(_in->read_exactly(to_read).get());
```

Comment:
I don"t understand. This requires a thread but the function returns a future.
Should it be a coroutine?